roles/foreman_installer/tasks/main.yml

```diff
@@ -0,0 +1,10 @@
+---
+- name: 'Run installer'
+  shell: >
+      {{ foreman_installer_command }}
+      --scenario {{ foreman_installer_scenario }}
```

Comment:
This variable is missing from defaults.

Reply:
I was thinking of keeping this out of defaults and having the role fail if you do not specify this. Rather than defaulting to Foreman and just assuming what the user wants.

Reply:
OK. I'd like to have a pre_task with "assert:" To provide a meaningful error message to the user.
(Not sure if you can add pre_tasks to roles)

```
pre_tasks:
  - name: Check that necessary variables are defined
    assert:
      that:
        - foreman_installer_scenario is defined
```

Reply:
Also the README should reflect that.

Reply:
Added into the role

Reply:
Why is `foreman` a bad default, tho? I mean, this is the foreman installer role, which kinda sounds like it's gonna install foreman by default? ;)

Reply:
to provide opinionated defaults? the project does not because it wants to inform the user how to get the list of scenarios, right (or are there other reasons)? the user won't use that role for it.

Reply:
I think the project does not because a user should declare what scenario they are wanting to install: foreman, katello, proxy, proxy with content. And I think, given there are many scenarios out there, a user should be forced to declare it explicitly. No surprises.